shared/utils/game_utils.py

```diff
@@ -284,6 +284,94 @@ def _check_single_king_and_throne(pieces: np.ndarray) -> bool:
     return True
 
 
+class Match(BaseModel):
+    """
+    Model representing a match in the Tablut game.
+
+    Attributes:
+        match_id (int): The unique identifier for the match.
+        white_player (AbstractPlayer): The player playing as white.
+        black_player (AbstractPlayer): The player playing as black.
+        turns (List[Turn]): The list of turns taken during the match.
+        states (List[State]): The list of states during the match.
+        actions (List[Action]): The list of actions taken during the match.
+        outcome (Turn): The outcome of the match.
+    """
+    from shared.utils import AbstractPlayer, State
+    match_id: int
+    white_player: AbstractPlayer
+    black_player: AbstractPlayer
+    turns: List[Turn]
+    states: List[State]
+    actions: List[Action]
```

Comment:
temove turns, actios and states and add tuns, a list of tuple each containing a State and an action. Because the turn of the action is already contained in Action class and because they have to be tracked and be accessed togheter